src/main/java/ch/vorburger/exec/ManagedProcess.java

```diff
@@ -473,7 +473,7 @@ public ManagedProcess waitForExitMaxMsOrDestroy(long maxWaitUntilDestroyTimeout)
     }
 
     protected void assertWaitForIsValid() throws ManagedProcessException {
-        if (!isAlive() && !resultHandler.hasResult()) {
+        if (!watchDog.isStopped() && !isAlive() && !resultHandler.hasResult()) {
```

Comment:
https://github.com/vorburger/ch.vorburger.exec/issues/99

Reply:
@vorburger it doesn't seem like this patch introduced this thread safety violation.  Can we merge in this patch?  It would be a useful improvement even if it preserved the thread safety issue.

Reply:
@mosesn thank you very much for having reviewed this; I agree - so (finally, sorry) MERGE this now!

Reply:
Thank you!!